iter/chain.go

```diff
@@ -14,6 +14,12 @@ func Chain[T any](iterators ...Iterator[T]) *ChainIter[T] {
 	return &ChainIter[T]{iterators, 0}
 }
 
+// Filter instantiates a [*FilterIter] that selectively yields only results
+// that cause the provided function to return `true`.
+func (iter *ChainIter[T]) Filter(fun func(T) bool) *FilterIter[T] {
+	return &FilterIter[T]{iter, fun, false}
```

Comment:
Use `Filter[T](iter, fun)` instead of manually declaring the struct - same for all uses of this in other iterators.